### PR TITLE
Edit Firebase from frameworks tech note

### PR DESCRIPTION
**Pull request by markarndt**
Cheers.

### CODE REVIEW

File: docs/firebase_in_libraries.md [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
-(especially when different versions of the static framework are linking to the
-app and the dynamic framework). For example, a `dispatch_once` may or may not do
-the right initialization since there are now two entities to initialize. Here
-are couple more examples of issues related to this undefined behavior:
+A dynamic framework is a bundle containing dynamic libraries and other resources. The dynamic
+libraries bundled in the framework can themselves be compiled from static libraries, like
+Firebase core and product libraries, meaning all of the symbols of the static libraries are
+part of the dynamic framework bundle. What if your core app already directly links a static
+Firebase library when you link to the same library indirectly from a dynamic framework? This
+leads to undefined behavior (especially when different versions of the static framework are
```
Comment (maksymmalyhin) [inline docs/firebase_in_libraries.md:18]: From the first glance it doesn't look exactly clear that the undefined behaviour is a result two(or more) copies of symbols are present in its runtime when such an application is launched. Do you think we should/can reword it to highlight this fact?

File: docs/firebase_in_libraries.md [hunk 2]
```diff
@@ -1,31 +1,28 @@
...
+linked to the app and the dynamic framework). For example, a `dispatch_once` may or may not
+perform the correct initialization since there are now two entities to initialize. Here are
+a couple more examples of issues related to this undefined behavior:
 [#4315](https://github.com/firebase/firebase-ios-sdk/issues/4315),
```
Comment (maksymmalyhin) [inline docs/firebase_in_libraries.md:22]: As suggested at b/154619039 we may add the exception message from issues like [this](https://stackoverflow.com/questions/60753233/the-default-firebaseapp-instance-must-be-configured-before-the-defaultfirebaseap) to make it searchable:
"The default FirebaseApp instance must be configured before the defaultFirebaseApp instance can be initialized".

File: docs/firebase_in_libraries.md [hunk 3]
```diff
@@ -1,31 +1,28 @@
...
-[#5152](https://github.com/firebase/firebase-ios-sdk/issues/4315), etc.
+[#5152](https://github.com/firebase/firebase-ios-sdk/issues/4315).
```
Comment (maksymmalyhin) [inline docs/firebase_in_libraries.md:23]: The link should be https://github.com/firebase/firebase-ios-sdk/issues/5152 ?

Comment (markarndt) [reply docs/firebase_in_libraries.md]: Git it.

Comment (maksymmalyhin) [reply docs/firebase_in_libraries.md]: Spotting my own errors 😄

File: docs/firebase_in_libraries.md (hunk 1 above)
Comment (markarndt) [reply docs/firebase_in_libraries.md]: Thanks!

File: docs/firebase_in_libraries.md (hunk 2 above)
Comment (markarndt) [reply docs/firebase_in_libraries.md:23]: Woops missed that from the gDoc. Got it!

Comment (maksymmalyhin) [reply docs/firebase_in_libraries.md:23]: Thanks!
